code/game/objects/items.dm

```diff
@@ -1010,3 +1010,159 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
+			if(S && S.tastes && S.tastes.len)
+				S.tastes += "bitterness"
+				S.tastes["bitterness"] = 3
+			found_mats++
+
+		if(found_mats < MAX_BONUS_MATS_PER_BITE && custom_materials[SSmaterials.GetMaterialRef(/datum/material/uranium)])
+			M.reagents.add_reagent(/datum/reagent/uranium, rand(4, 6))
+			found_mats++
+
+		if(found_mats < MAX_BONUS_MATS_PER_BITE && custom_materials[SSmaterials.GetMaterialRef(/datum/material/bluespace)])
+			M.reagents.add_reagent(/datum/reagent/bluespace, rand(5, 8))
+			if(S && S.tastes && S.tastes.len)
+				S.tastes += "blue"
+				S.tastes["blue"] = 3 //tastes like... blue
+			found_mats++
+
+		if(found_mats < MAX_BONUS_MATS_PER_BITE && custom_materials[SSmaterials.GetMaterialRef(/datum/material/bananium)])
+			if(S && S.tastes && S.tastes.len)
+				S.tastes += "banana"
+				S.tastes["banana"] = 3
+			found_mats++
+
+		if(found_mats < MAX_BONUS_MATS_PER_BITE && custom_materials[SSmaterials.GetMaterialRef(/datum/material/hot_ice)])
+			M.reagents.add_reagent(/datum/reagent/toxin/plasma, rand(5, 6)) 
+			if(S && S.tastes && S.tastes.len)
+				S.tastes += "salt"
+				S.tastes["salt"] = 3
+			found_mats++
+
+		//Primary mats (glass, iron, etc) in order of hardness (something with both iron and glass probably won't shatter when you bite into it)
+
+		if(custom_materials[SSmaterials.GetMaterialRef(/datum/material/adamantine)] || custom_materials[SSmaterials.GetMaterialRef(/datum/material/runite)])
+			M.apply_damage(20, BRUTE, BODY_ZONE_HEAD, wound_bonus = 10) //oof my tooth
+
+		else if(custom_materials[SSmaterials.GetMaterialRef(/datum/material/diamond)] || custom_materials[SSmaterials.GetMaterialRef(/datum/material/titanium)])
+			M.apply_damage(15, BRUTE, BODY_ZONE_HEAD, wound_bonus = 7)
+
+		else if(custom_materials[SSmaterials.GetMaterialRef(/datum/material/iron)])
+			M.apply_damage(10, BRUTE, BODY_ZONE_HEAD, wound_bonus = 5) 
+
+		else if(custom_materials[SSmaterials.GetMaterialRef(/datum/material/wood)])
+			M.apply_damage(5, BRUTE, BODY_ZONE_HEAD)
+			if(S && S.tastes && S.tastes.len)
+				S.tastes += "wood chips and sawdust"
+				S.tastes["wood chips and sawdust"] = 3
+
+		else if(custom_materials[SSmaterials.GetMaterialRef(/datum/material/glass)])
+			M.apply_damage(10, BRUTE, BODY_ZONE_HEAD, wound_bonus = 5, sharpness = TRUE) //cronch
+			//biting glass doesn't mean you swallow the glass, but it DOES leave the shard in there if you keep eating like an idiot
+			/// The glass shard that is spawned into the source item
+			var/obj/item/shard/broken_glass = null
+
+			if(istype(src, /obj/item/reagent_containers)) //if it's a glass reagent container...
+				if(reagents.reagent_list && reagents.reagent_list.len)
+					for(var/P in reagents.reagent_list)
+						if(istype(P, /datum/reagent/toxin/plasma) || istype(P, /datum/reagent/stable_plasma))
+							broken_glass = new /obj/item/shard/plasma(src) //if there's plasma in the beaker, make a plasma glass shard

```

Comment:
if inserting a beaker with 1u of plasmaglass in it into a loaf of bread, then biting it becomes the new meta way to rush plasmaglass spear creation, I'm gonna laugh so hard

eh, this is probably fine, even if it doesn't make too much sense if you think about it too hard (it's funny, and that's what counts)